src/elements.tsx

```diff
@@ -4,6 +4,14 @@
 
 import * as os from 'os';
 
+export interface Attributes {
+    [key: string]: AttributeValue;
+}
+
+type AttributeValue = number | string | Date | boolean;
+
+type CustomElement = { (attributes: Attributes | undefined, contents: string[]): string }
```

Comment:
Great job naming it separate here. What do you think of the name `CustomElementHandler` instead? It more clearly communicates the intent of the type.

Could we also make it into an `interface` instead of a `type`? It is the convention to use interfaces if we can.